cpp/src/io/utilities/parsing_utils.cuh

```diff
@@ -128,6 +128,35 @@ __device__ __forceinline__ uint8_t decode_digit(char c, bool* valid_flag)
   return 0;
 }
 
+// Converts character to lowercase.
+__inline__ __device__ char to_lower(char const c)
+{
+  return c >= 'A' && c <= 'Z' ? c + ('a' - 'A') : c;
+}
+
+/**
+ * @brief Check if string is infinity, case insensitive with/without sign
+ * Valid infinity strings are inf, +inf, -inf, infinity, +infinity, -infinity
+ * String comparison is case insensitive.
+ *
+ * @param data The character string for parse
+ * @param start The index within data to start parsing from
+ * @param end The end index within data to end parsing
+ * @return true if string is valid infinity, else false.
+ */
+__inline__ __device__ bool is_infinity(const char* data, long start, long end)
```

Comment:
I know that this is a late time to be gathering requirements, but Spark supports two configs to let the user set the value for positive infinity and negative infinity.  It would be great to be able to support something similar.

If it is too late for that it is not a big deal because at least this will support parsing the default values for those configs.

Reply:
@revans2 sounds like we could support that specifically in spark using a `trie` and move it in to util as required. Is there a feature request for this?

Reply:
I can file a feature request if needed.  For now it has been a documented incompatibility, but I noticed this go by so I thought I would ask.

Reply:
File a feature request. This feature is blocked by PR #6122.

Reply:
FWIW, we already use tries to check for special NaN and true/false values: https://github.com/rapidsai/cudf/blob/branch-0.16/cpp/src/io/csv/csv_gpu.cu#L223-L227
So using one here out of the line in terms of the overhead.